openshift_cli_installer/libs/clusters/aro_cluster.py

```diff
@@ -0,0 +1,248 @@
+        with open(os.path.join(auth_path, "kubeconfig"), "w") as kubeconfig:
+            kubeconfig.write(self.aro_cluster_kubeconfig())
+
+        with open(os.path.join(auth_path, "kubeadmin-password"), "w") as kubeadmin:
+            kubeadmin.write(self.aro_cluster_kubeadmin_password())
+
+    def create_cluster(self):
+        self.create_cluster_resources()
+        self.timeout_watch = self.start_time_watcher()
+
+        cluster_body = {
+            "clusterProfile": {
+                "domain": self.cluster["domain"],
+                "fipsValidatedModules": "Enabled" if self.cluster["fips"] else "Disabled",
+                "pullSecret": self.pull_secret,
+                "resourceGroupId": f"/subscriptions/{self.subscription_id}/resourcegroups/{self.cluster['cluster-resource-group-name']}",
+                "version": self.cluster["version"],
+            },
+            "masterProfile": {
+                "encryptionAtHost": "Enabled",
+                "subnetId": f"/subscriptions/{self.subscription_id}/resourceGroups/{self.resource_group_name}/providers/Microsoft"
+                f".Network/virtualNetworks/{self.virtual_network_name}/subnets/{self.cluster['master-subnet-name']}",
+                "vmSize": self.cluster["master-vm-size"],
+            },
+            "workerProfiles": [
+                {
+                    "count": self.cluster["workers-count"],
+                    "diskSizeGB": self.cluster["workers-disk-size"],
+                    "name": "worker",
+                    "encryptionAtHost": "Enabled",
+                    "subnetId": f"/subscriptions/{self.subscription_id}/resourceGroups/{self.resource_group_name}/providers/Microsoft"
+                    f".Network/virtualNetworks/{self.virtual_network_name}/subnets/{self.cluster['workers-subnet-name']}",
+                    "vmSize": self.cluster["workers-vm-size"],
+                }
+            ],
+            "servicePrincipalProfile": {
+                "clientId": self.client_id,
+                "clientSecret": self.client_secret,
+            },
+            "apiserverProfile": {"visibility": "Public"},
+            "consoleProfile": {},
+            "ingressProfiles": [{"name": "default", "visibility": "Public"}],
+            "networkProfile": {
+                "podCidr": self.cluster["network-pod-cidr"],
+                "preconfiguredNSG": "Disabled",
+                "serviceCidr": self.cluster["network-service-cidr"],
+            },
+        }
+
+        self.logger.info(f"Creating {self.platform} cluster {self.cluster_name}")
+        self.aro_client.open_shift_clusters.begin_create_or_update(
+            resource_name=self.cluster_name,
+            resource_group_name=self.resource_group_name,
+            parameters={"location": self.region, "properties": cluster_body},
+        ).result()
+        self.set_aro_cluster_auth()
+
+        self.logger.info(f"{self.platform} cluster {self.cluster_name} created successfully.")
```

Comment:
if fail; we need to rollback

Reply:
done, deleting cluster + resources on failure